### PR TITLE
Update dependency go modules for k8s v1.28.0-rc.0

### PR DESCRIPTION
**Pull request by Sneha-at**
Ran kubernetes-csi/csi-release-tools go-get-kubernetes.sh -p 1.28.0-rc.0

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,67 +5,67 @@ go 1.18
...
-	golang.org/x/net v0.8.0
-	google.golang.org/grpc v1.51.0
-	k8s.io/api v0.27.0
-	k8s.io/client-go v0.27.0
+	github.com/stretchr/testify v1.8.2
+	golang.org/x/net v0.12.0
+	google.golang.org/grpc v1.54.0
+	k8s.io/api v0.28.0-rc.0
+	k8s.io/client-go v0.28.0-rc.0
 	k8s.io/component-base v0.26.0
```
Comment (sunnylovestiramisu) [inline go.mod:13]: This one should be k8s.io/component-base v0.28.0-rc.0

Comment (xing-yang) [reply go.mod:13]: +1

**Pull request merged**